cloudevents/generic/options/grpc/options.go

```diff
@@ -0,0 +1,130 @@
+package grpc
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"open-cluster-management.io/api/cloudevents/generic/options/grpc/protocol"
+)
+
+const (
+	// SpecTopic is a pubsub topic for resource spec.
+	SpecTopic = "sources/+/clusters/+/spec"
+
+	// StatusTopic is a pubsub topic for resource status.
+	StatusTopic = "sources/+/clusters/+/status"
+
+	// SpecResyncTopic is a pubsub topic for resource spec resync.
+	SpecResyncTopic = "sources/clusters/+/specresync"
+
+	// StatusResyncTopic is a pubsub topic for resource status resync.
+	StatusResyncTopic = "sources/+/clusters/statusresync"
+)
+
+type GRPCOptions struct {
+	Host           string
+	Port           int
+	CAFile         string
+	ClientCertFile string
+	ClientKeyFile  string
+}
+
+func NewGRPCOptions() *GRPCOptions {
+	return &GRPCOptions{}
+}
+
+func (o *GRPCOptions) AddFlags(flags *pflag.FlagSet) {
```

Comment:
I think this should also be in the config similar as mqtt. It will be better it we share the same config file.

Reply:
updated the grpc options to use config similar to mqtt.